packages/react/types/index.d.ts

```diff
@@ -46,14 +46,12 @@ export function withEmotionCache<Props, RefType = any>(
   func: (props: Props, context: EmotionCache, ref: Ref<RefType>) => ReactNode
 ): FC<Props & ClassAttributes<RefType>>
 
-type ReactInterpolationPrimitive =
-  | null
-  | undefined
-  | boolean
-  | SerializedStyles
-  | CSSObject
+type Falsy = false | '' | 0 | null | undefined
+type ReactInterpolationPrimitive = Falsy | SerializedStyles | CSSObject
 
-type ReactArrayInterpolation<Props> = Array<ReactInterpolation<Props>>
+type ReactArrayInterpolation<Props> = Array<
+  ReactInterpolationPrimitive | ReactFunctionInterpolation<Props>
+>
```

Comment:
Any reason why this has been split? An additional problem this has that this doesn't allow nested arrays now.

Reply:
Yes so it was creating a cycle with type alias so I had to _break_ the cycle to allow the tests to pass.

But I think as you noticed here: https://github.com/emotion-js/emotion/pull/2296#issuecomment-801705893 will allow avoiding these changes since TS 3.7 brought a change that allows this pattern.